examples/decoding/decoding_rsa.py

```diff
@@ -0,0 +1,191 @@
+# Epoch data
+picks = mne.pick_types(raw.info, meg=True)
+epochs = mne.Epochs(raw, events=events, event_id=event_id, baseline=None,
+                    picks=picks, tmin=-.1, tmax=.500, preload=True)
+
+##############################################################################
+# Let's plot some conditions
+epochs['face'].average().plot()
+epochs['not-face'].average().plot()
+
+##############################################################################
+# Representational Similarity Analysis (RSA) is a neuroimaging-specific
+# appelation to refer to statistics applied to the confusion matrix
+# also referred to as the representational dissimilarity matrices (RDM).
+#
+# Compared to the approach from Cichy et al. we'll use a multiclass
+# classifier (Multinomial Logistic Regression) while the paper uses
+# all pairwise binary classification task to make the RDM.
+# Also we use here the ROC-AUC as performance metric while the
+# paper uses accuracy. Finally here for the sake of time we use
+# RSA on a window of data while Cichy et al. did it for all time
+# instants separately.
+
+# Classify using the average signal in the window 50ms to 300ms
+clf = make_pipeline(StandardScaler(),
+                    LogisticRegression(C=1, solver='lbfgs'))
+X = epochs.copy().crop(0.05, 0.3).get_data().mean(axis=2)
+y = epochs.events[:, 2]
+
+n_folds = 5
+classes = set(y)
+cv = StratifiedKFold(n_folds, random_state=0, shuffle=True)
+
+# Compute confusion matrix for each cross-validation fold
+y_pred = np.zeros((len(y), len(classes)))
+for train, test in cv.split(X, y):
+    # Fit
+    clf.fit(X[train], y[train])
+    # Predict
+    y_pred[test] = clf.predict_proba(X[test])
+
+print(y_pred[:2])
+
+
+##############################################################################
+# Compute confusion matrix using ROC-AUC
+cm = np.zeros((len(classes), len(classes)))
+for ii, train_class in enumerate(classes):
+    for jj in range(ii, len(classes)):
+        cm[ii, jj] = roc_auc_score(y == train_class, y_pred[:, jj])
+        cm[jj, ii] = cm[ii, jj]
+
+# Format class names for centered plotting
+names = np.array([conds[conds.trigger == (val - 1)].condition.iloc[0]
+                  for val in classes])
+sparse_names = np.copy(names)
+n = 0
+for ii, name in enumerate(names):
+    sparse_names[ii] = '' if n != (sum(names == name) // 2) else name
+    n = 0 if ii < (len(names) - 1) and names[ii + 1] != name else n + 1
+
+##############################################################################
+# Plot
+fig, ax = plt.subplots(1)
+im = ax.matshow(cm, cmap='RdBu_r', clim=[0.3, 0.7])
+ax.set_yticks(range(len(classes)))
+ax.set_yticklabels(sparse_names)
+ax.set_xticks(range(len(classes)))
+ax.set_xticklabels(sparse_names, rotation=40, ha='left')
+for ii, (name, next_name) in enumerate(zip(names, names[1:])):
+    if name != next_name:
+        ax.axhline(ii + 0.5, color='k')
+        ax.axvline(ii + 0.5, color='k')
```

Comment:
same here, I would simplify to a manual loop, no need to be too smart with that number of categories. (my bad)